src/resolve-url.js

```diff
@@ -7,12 +7,38 @@ const resolveUrl = (baseUrl, relativeUrl) => {
     return relativeUrl;
   }
 
+  // IE11 supports URL but not the URL constructor
+  // feature detect the behavior we want
+  const nativeURL = typeof window.URL === 'function';
+
+  const protocolLess = (/^\/\//.test(baseUrl));
+  // remove location if window.location isn't available (i.e. we're in node)
+  // and if baseUrl isn't an absolute url
+  const removeLocation = !window.location && !(/\/\//i).test(baseUrl);
+
   // if the base URL is relative then combine with the current location
-  if (!(/\/\//i).test(baseUrl)) {
+  if (nativeURL) {
+    baseUrl = new window.URL(baseUrl, window.location || 'http://example.com');
+  } else if (!(/\/\//i).test(baseUrl)) {
     baseUrl = URLToolkit.buildAbsoluteURL(window.location && window.location.href || '', baseUrl);
   }
 
+  if (nativeURL) {
+    const newUrl = new URL(relativeUrl, baseUrl);
+
+    // if we're a protocol-less url, remove the protocol
+    // and if we're location-less, remove the location
+    // otherwise, return the url unmodified
+    if (removeLocation) {
+      return newUrl.href.slice(18);
```

Comment:
Probably worth adding `const DEFAULT_LOCATION = 'http://example.com'` and DEFAULT_LOCATION.length